### PR TITLE
Update android studio

### PR DESCRIPTION
**Pull request by daneov**
## Description of changes

Android Studio has received a patch update, and the beta/canary versions have furthered as well.
This PR brings them up to the current version, as described in the commit messages.

I haven't had the time yet to test them, but plan on doing so soon.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html)) - Using NixOS, enabled the config setting.
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by daneov** [comment]
Todo:

- [ ] Update the hashes
- [ ] Test build result

**Comment by daneov** [comment]
Tagging @alapshin as maintainer.

**Review by alapshin — APPROVED** [review]

File: pkgs/applications/editors/android-studio/default.nix [hunk 1]
```diff
@@ -10,16 +10,16 @@ let
...
-    version = "2022.3.1.17"; # "Android Studio Giraffe (2022.3.1) RC 1"
-    sha256Hash = "sha256-Bpur9ytiwa0udiyYVxlfLT4M+ZcO5atQUQg/ForATP4=";
+    version = "2023.1.1.17"; # "Android Studio Hedgehog (2023.3.1)"
+    sha256Hash = "sha256-0sN+B1RxxlbgxXrEs8gft4qjvIYtazJS6DllHZ2h768=";
   };
   latestVersion = {
-    version = "2023.1.1.14"; # Android Studio Hedgehog (2023.1.1) Canary 14
-    sha256Hash = "sha256-MgZDUpRyM0XDfgqfYjtJJyG2CBaNB06PgtoJltDwNwk=";
+    version = "2023.2.1.1"; # Android Studio Iguana (2023.2.1) Canary 1
+    sha256Hash = "sha256-7ub9GkJd9J37nG2drXOuU7r4w/gI+m9nlWANqIk2ddk=";
```
Comment (Janik-Haag) [inline pkgs/applications/editors/android-studio/default.nix:22]: Seems like there is a mistake with the  Canary version in the comment shouldn't it be 15?

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 251840` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>android-studio (androidStudioPackages.stable)</li>
    <li>androidStudioPackages.beta</li>
    <li>androidStudioPackages.canary</li>
    <li>androidStudioPackages.dev</li>
  </ul>
</details>

File: pkgs/applications/editors/android-studio/default.nix (hunk 1 above)
Comment (daneov) [reply pkgs/applications/editors/android-studio/default.nix:22]: Not as far as I understand: `Giraffe` went to stable, `Hedgehog` to beta and `Iguana` became canary.

With `Iguana` the minor was incremented and patch was reset to 1:

image

Source: https://archive.ph/BNZik

Comment (Janik-Haag) [reply pkgs/applications/editors/android-studio/default.nix:22]: Okay fair enough

**Review by Janik-Haag — APPROVED** [review]

**Pull request merged**